### ISSUE TITLE
sherpa calc_ftest vs. Xspec ftest

### ISSUE THREAD

**Issue by dtnguyen2** [sherpa/sherpa#641]
The original bug report can be found here:
https://icxc.harvard.edu/pipe/ascds_help/2019/0254.html

The reply to the bug report can be found here:
https://icxc.harvard.edu/pipe/ascds_help/2019/0255.html

# Note

I haven't looked for the specs on calc_ftest, but judging from the docs:
```
    Examples                                                                    
    --------                                                                    
                                                                                
    In this example, the simple model has 5 degrees of freedom and              
    a chi-square statistic of 7.73, while the complex model has 8               
    degrees of freedom and a chi-square statistic of 8.94. The                  
    F test does not provide any evidence that the complex model                 
    is a better fit to the data than the simple model since the                 
    result is much larger than 0.                                               

    >>> calc_ftest(5, 7.73, 8, 8.94)                                            
    0.32480691622314933
```                                                         

The current code is consistent with the docs:

```
Python 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sherpa.astro.ui import *
>>> calc_ftest(5, 7.73, 8, 8.94)
0.32480691622314933
```

I assume someone must have looked at the docs and approved it; Moreover, the doc has the following entries:

```
    Parameters
    ----------
    dof1 : int
       degrees of freedom of the simple model
    stat1 : number
       best-fit chi-square statistic value of the simple model
    dof2 : int
       degrees of freedom of the complex model
    stat2 : number
       best-fit chi-square statistic value of the complex model
```
which is not true since one can enter the args as arrays/lists/tuples, for example (note, I have already convert the calculations to follow XSPEC):
```
>>> calc_ftest([11, 11], [16.3, 16.3], [10, 9], [10.2, 10.5]) 
array([0.03452353, 0.13819987])
```

**Comment by anetasie** [comment]
The example in the online docs does not look correct, because the more complex model will have a lower number degrees of freedom. 
Our web site provide the example with the expected input:
[http://cxc.harvard.edu/sherpa/ahelp/calc_ftest.html]

I did not know that we can use the arrays as an input to this function.

**Comment by dtnguyen2** [comment]
Using the code in PR #643:
```
>>> calc_ftest(5, 7.73, 8, 8.94)
nan
>>> calc_ftest(2, 20.28, 34, 33.63)
nan
>>> calc_ftest(11, 16.3, 10, 10.2)
0.03452352914891555
```
Where the first entry is the example in the current version of sherpa doc for calc_ftest, the second is from the provide example in [http://cxc.harvard.edu/sherpa/ahelp/calc_ftest.html] and the last is from the bug report (which now agrees with XSPEC's result).  As I said in the PR, if you don't think we should get the same answer as XSPEC then reject PR #643.

**Comment by dtnguyen2** [comment]
At first, I thought perhaps XSPEC's definitions of chi1 and dof1 is sherpa's definitions of chi2 and dof2 etc... So I decided to reread XSPEC code for ftest; However, I noticed that XSPEC requires that dof2 < dof1:

```
   if (isOK)
   {
      Real& chi2 = vals[0];
      Real& dof2 = vals[1];
      Real& chi1 = vals[2];
      Real& dof1 = vals[3];
      if ( chi2 <= 0 || dof1 <= dof2)
      {
         tcerr << "Bad values for F-Test: \n";
         if ( chi2 <= 0 ) tcerr << "\t Chi2 = " << chi2 << '\n';
         if (dof1 <= dof2) tcerr << "\t dof1 <= dof2 \n";       
         if (dof2*dof1 == 0) tcerr << "\t  degrees of freedom are zero: "
                         << " 1: " << dof1 << " 2: " << dof2 << '\n';     
         return -1;
      }
```
sherpa has no such requirement:
```
>>> calc_ftest(11, 16.3, 10, 10.2)
0.03452352914891555
>>> calc_ftest(10, 10.2, 11, 16.3)
nan
```

**Comment by DougBurke** [comment]
This has been addressed in #704
